src/libudev/meson.build

```diff
@@ -16,11 +17,57 @@ libudev_sources = files('''
 
 ############################################################
 
+libudev_includes = [includes, include_directories('.')]
+
 libudev_sym = files('libudev.sym')
-libudev_sym_path = meson.current_source_dir() + '/libudev.sym'
+libudev_sym_path = join_paths(meson.current_source_dir(), 'libudev.sym')
 
 install_headers('libudev.h')
-libudev_h_path = '@0@/libudev.h'.format(meson.current_source_dir())
+libudev_h_path = join_paths(meson.current_source_dir(), 'libudev.h')
+
+libudev_basic = static_library(
+        'udev-basic',
+        libudev_sources,
+        include_directories : includes,
+        c_args : ['-fvisibility=default'])
+
+libudev_static = static_library(
+        'udev_static',
+        include_directories : includes,
+        link_with : udev_link_with,
+        link_whole : libudev_basic)
+
+static_libudev = get_option('static-libudev')
+static_libudev_pic = static_libudev == 'true' or static_libudev == 'pic'
+install_libudev_static = static_library(
+        'udev',
+        basic_sources,
+        shared_sources,
+        libsystemd_sources,
+        libudev_sources,
+        disable_mempool_c,
+        include_directories : includes,
+        build_by_default : static_libudev != 'false',
+        install : static_libudev != 'false',
+        install_dir : rootlibdir,
+        link_depends : libudev_sym,
+        dependencies : libshared_deps + [libmount],
+        c_args : static_libudev_pic ? [] : ['-fno-PIC'],
+        pic : static_libudev_pic)
+
+libudev = shared_library(
```

Comment:
Hmm, I think the location of libudev.so is referenced in some external places. I vaguely remember @poettering having to add some directories to LD_LIBRARY_PATH a few months ago. If we're moving this, I'd prefer to move the .so file to build/, so that we don't need to play games with subdirectories in LD_LIBRARY_PATH.

Reply:
OK. Now, libudev is also built in the build directory.